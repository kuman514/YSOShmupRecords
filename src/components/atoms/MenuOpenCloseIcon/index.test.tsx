import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { beforeEach, describe, expect, it } from 'vitest';
import 'jest-styled-components';

import { MenuOpenCloseIcon } from '.';

describe('MenuButtonIcon', () => {
  beforeEach(() => {
    cleanup();
  });

  it('should be a stack icon when closed', () => {
    const { container } = render(<MenuOpenCloseIcon isOpen={false} />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstElementChild?.id).toStrictEqual('Click-to-open-icon');
  });

  it('should be a cross icon when open', () => {
    const { container } = render(<MenuOpenCloseIcon isOpen />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstElementChild?.id).toStrictEqual(
      'Click-to-close-icon'
    );
  });
});
