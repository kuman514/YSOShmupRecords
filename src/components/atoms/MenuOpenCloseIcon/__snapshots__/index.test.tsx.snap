// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MenuButtonIcon > should be a cross icon when open 1`] = `
.c1 {
  width: 24px;
  height: 2px;
  background-color: #ffffff;
  transition-property: left,top,transform;
  transition-duration: 500ms;
  transition-timing-function: ease-in-out;
  position: absolute;
  left: 6px;
  top: 17px;
  transform: rotate(45deg);
}

.c2 {
  width: 24px;
  height: 2px;
  background-color: #ffffff;
  transition-property: left,top,transform;
  transition-duration: 500ms;
  transition-timing-function: ease-in-out;
  position: absolute;
  left: 6px;
  top: 17px;
  transform: rotate(-45deg);
}

.c3 {
  width: 24px;
  height: 2px;
  background-color: #ffffff;
  transition-property: left,top,transform;
  transition-duration: 500ms;
  transition-timing-function: ease-in-out;
  position: absolute;
  left: 6px;
  top: 17px;
  transform: rotate(-45deg);
}

.c0 {
  width: 36px;
  height: 36px;
  position: relative;
}

<div
  class="c0"
  id="Click-to-close-icon"
>
  <div
    class="c1"
  />
  <div
    class="c2"
  />
  <div
    class="c3"
  />
</div>
`;

exports[`MenuButtonIcon > should be a stack icon when closed 1`] = `
.c1 {
  width: 24px;
  height: 2px;
  background-color: #ffffff;
  transition-property: left,top,transform;
  transition-duration: 500ms;
  transition-timing-function: ease-in-out;
  position: absolute;
  left: 6px;
  top: 9px;
  transform: rotate(0deg);
}

.c2 {
  width: 24px;
  height: 2px;
  background-color: #ffffff;
  transition-property: left,top,transform;
  transition-duration: 500ms;
  transition-timing-function: ease-in-out;
  position: absolute;
  left: 6px;
  top: 17px;
  transform: rotate(0deg);
}

.c3 {
  width: 24px;
  height: 2px;
  background-color: #ffffff;
  transition-property: left,top,transform;
  transition-duration: 500ms;
  transition-timing-function: ease-in-out;
  position: absolute;
  left: 6px;
  top: 25px;
  transform: rotate(0deg);
}

.c0 {
  width: 36px;
  height: 36px;
  position: relative;
}

<div
  class="c0"
  id="Click-to-open-icon"
>
  <div
    class="c1"
  />
  <div
    class="c2"
  />
  <div
    class="c3"
  />
</div>
`;
